core/src/main/scala/kafka/utils/PasswordEncoder.scala

```diff
@@ -130,7 +131,7 @@ class EncryptingPasswordEncoder(
       val decrypted = cipher.doFinal(encryptedPassword)
       new String(decrypted, StandardCharsets.UTF_8)
     } catch {
-      case e: Exception => throw new ConfigException("Password could not be decoded", e)
+      case e: Exception => throw new InvalidConfigurationException("Password could not be decoded", e)
```

Comment:
took a quick look at usages -- looks like dynamic broker configs where it is caught and handled and the ZK migration code that doesn't seeme to specifically handle either error case.